share/spice/dota2/dota2.js

```diff
@@ -0,0 +1,148 @@
+            hero = herodata[heroName];
+            hero.heroKey = heroName;
+        } else {
+            $.each(herodata, function(index, value) {
+                if (value.dname.toLowerCase() == heroName) {
+                    hero = value;
+                    hero.heroKey = index;
+                }
+            });
+        }
+        return hero;
+    }
+
+    function getHeroRoles(hero) {
+        return hero.droles.replace(/ - /g, ', ');
+    }
+
+    function getHeroAbilities(hero, result) {
+        var abilities = [];
+        $.each(result.abilitydata, function(index, value) {
+            if (index.indexOf(hero.heroKey) === 0) {
+                abilities.push(
+                    {
+                        'name': value.dname,
+                        'image': 'http://cdn.dota2.com/apps/dota2/images/abilities/' + index + '_hp1.png'
+                    }
+                );
+            }
+        });
+        return abilities;
+    }
+
+    function getHeroPrimaryAttribute(hero) {
+        switch (hero.pa) {
+            case 'int':
+                return 'Intelligence';
+            case 'str':
+                return 'Strength';
+            case 'agi':
+                return 'Agility';
+        }
+    }
+
+    function createHeroReturnObject(hero, result) {
+        return {
+            isHero: true,
+            name: hero.dname,
+            image: 'http://cdn.dota2.com/apps/dota2/images/heroes/' + hero.heroKey + '_vert.jpg',
+            attackType: hero.dac,
+            primaryAttribute: getHeroPrimaryAttribute(hero),
+            int: hero.attribs.int.b + ' + ' + hero.attribs.int.g,
+            agi: hero.attribs.agi.b + ' + ' + hero.attribs.agi.g,
+            str: hero.attribs.str.b + ' + ' + hero.attribs.str.g,
+            attack: hero.attribs.dmg.min + ' - ' + hero.attribs.dmg.max,
+            speed: hero.attribs.ms,
+            defense: hero.attribs.armor,
+            roles: getHeroRoles(hero),
+            abilities: getHeroAbilities(hero, result)
+        };
+    }
+
+    env.ddg_spice_dota2 = function(api_result){
+
+        if (api_result.error) {
+            return Spice.failed('dota2');
+        }
+
+        var searchTarget = DDG.get_query().replace("dota2", "").replace("dota 2", "").trim().toLowerCase();
+        var returnData;
+        var item = getItemFromData(api_result.itemdata, searchTarget);
+        var sourceUrl;
+        if (typeof item !== 'undefined') {
+            returnData = createItemReturnObject(item, api_result);
+            sourceUrl = 'http://www.dota2.com/items/';
+        } else {
+            var hero = getHeroFromData(api_result.herodata, searchTarget);
+            if (typeof hero !== 'undefined') {
+                returnData = createHeroReturnObject(hero, api_result);
+            }
+            sourceUrl = 'http://www.dota2.com/hero/' + hero.dname.replace(" ", "_");
+        }
+
+        Spice.add({
+            id: "dota2",
+            name: "Dota2",
```

Comment:
Let's change this to "Games" so it corresponds with on of our existing Instant Answer Topics